stronk/__init__.py

```diff
@@ -52,16 +53,20 @@
 app.register_blueprint(workouts_page, url_prefix='/workouts')
 app.register_blueprint(exercise_page, url_prefix='/exercises')
 
+app.before_request(auth.verify_token)
+
 # a simple page that says hello
 @app.route('/')
 def index():
     return 'Hello, World!'
 
+
 """ graphql route
 this uses the flask as_view utility which transforms a class to a view function, passing its args to the class constructor
 each request will call the class' `dispatch_request()` function
 
 - flask [as_view](https://flask.palletsprojects.com/en/1.1.x/api/#flask.views.View.as_view)
 - flask-graphql  [as_view](https://github.com/graphql-python/flask-graphql/blob/master/flask_graphql/graphqlview.py)
 """
-app.add_url_rule('/graphql', view_func=GraphQLView.as_view('graphql',schema=schema,graphiql=app.debug))
+app.add_url_rule('/graphql', view_func=GraphQLView.as_view('graphql',
```

Comment:
I think we should add another route strictly for accessing the graphiiql interface since we have the `before_request` for this route. Maybe something like:
```
# actual graphql endpoint
app.add_url_rule('/graphql', view_func=GraphQLView.as_view('graphql',schema=schema))

# debug endpoint for graphiql
if (app.debug):
    app.add_url_rule('/graphiql', view_func=GraphQLView.as_view('graphiql',schema=schema,graphiql=true))
```

Reply:
Also, maybe not in scope, but maybe we can add a basic error handler?

```python
@app.errorhandler(Exception)
def handleGenericError(error):
    return jsonify(error=error.__class__.__name__, status=404)
```

Reply:
@13ABEL sounds good, I'm thinking of making it a 500 though since it's an internal server problem for more generic errors and not a resource not found error. Thoughts?

Reply:
o ya 404 is just placeholder haha. Our error classes should prob include the error status